buildkitd/certificates.go

```diff
@@ -247,3 +249,58 @@ func savePEM(path, typ string, bytes []byte) error {
 
 	return nil
 }
+
+// ConfigureSatelliteTLS uses the CA cert and key associate with the satellite
+// to generate a new certificate/key pair for use in client-side mTLS.
+// The certificates are configured in the settings for a new buildkit client.
+// The certificates are also stored on disk in: `~/.earthly/certs/<org-id>/<sat-name>`.
+func ConfigureSatelliteTLS(sat *cloud.SatelliteInstance, settings *Settings, cfg *config.Config) error {
+	dir := filepath.Join(cfg.Global.SatelliteCertsDir, sat.Org, sat.Name)
+	caCertPath := filepath.Join(dir, "ca_cert.pem")
+	caKeyPath := filepath.Join(dir, "ca_key.pem")
+	earthlyCertPath := filepath.Join(dir, "earthly_cert.pem")
+	earthlyKeyPath := filepath.Join(dir, "earthly_key.pem")
+
+	settings.TLSCA = caCertPath
+	settings.ClientTLSCert = earthlyCertPath
+	settings.ClientTLSKey = earthlyKeyPath
+	settings.ServerTLSCert = ""
+	settings.ServerTLSKey = ""
+
+	if err := os.RemoveAll(dir); err != nil {
```

Comment:
will this error if the previous certificates don't exist?

Reply:
It works when the dir doesn't exist (in fact they commented that it returns nil in that case in the src)

Reply:
this isn't needed now that it's a temp dir; however, we should have a deferred call to remove the temp dir before earthly exits.

I wouldn't worry about the edge-case of someone doing a hard-kill of earthly, the temp dir will eventually get cleaned by the OS..... I hope at least :)

Reply:
Okay I'll look into the deferred approach.  I'm also wondering now if the path contain a UUID, so two different earthly processes don't try to write/remove certs for the same satellite (they would each generated different client certs)

Reply:
I've gone with the approach of having the new `ConfigureSatelliteTLS` function return a cleanup func that can be deferred by the caller. 

I have this cleanup func also returned through `GetBuildkitClient`, which is used in many places like debug commands. I think its a bit ugly to return both a cleanupTLS function and a client that needs to be closed here. It could be more graceful if that cleanup happened automatically as part of the buildkit client's `Close()`, but that seemed like too much refactoring to be worth it.

WDYT @alexcb ?

Reply:
The solution you have looks good (just left one minor suggestion)